SetupDataPkg/ConfDataSettingProvider/ConfDataSettingProvider.c

```diff
@@ -294,6 +294,179 @@ DFCI_SETTING_PROVIDER  mSettingsProvider = {
+    /*
+     * Var List is in DmpStore format:
+     * 
+     *  struct {
+     *    RUNTIME_VAR_LIST_HDR VarList;
+     *    char Name[VarList->NameSize];
+     *    char Guid[16];
+     *    UINT32 Attributes;
+     *    char Data[VarList->DataSize];
+     *    UINT32 CRC32; // CRC32 of all bytes from VarList to end of Data
+     *  }
+     */
+    name = (CHAR16 *)(VarList + 1);
+    Guid = (EFI_GUID *)((CHAR8 *)name + VarList->NameSize);
+    Attributes = *(UINT32 *)(Guid + 1);
+    Data = (CHAR8 *)Guid + sizeof(*Guid) + sizeof(Attributes);
+    CRC32 = *(UINT32 *)(Data + VarList->DataSize);
+    LenToCRC32 = sizeof(*VarList) + VarList->NameSize + VarList->DataSize + sizeof(*Guid) + sizeof(Attributes);  
+
+    // on next iteration, skip past this variable
+    i += LenToCRC32 + sizeof(CRC32);
+
+    // validate CRC32
+    CalcCRC32 = 0;
+    Status = gBS->CalculateCrc32(VarList, LenToCRC32, &CalcCRC32); 
+
+    if (EFI_ERROR(Status) || CalcCRC32 != CRC32) {
+      // Either CRC calulation failed or CRC didn't match, drop this variable, continue to the next
+      DEBUG ((DEBUG_ERROR, "DFCI Runtime Settings Provider failed CRC check, skipping applying setting: %s Status: %r Received CRC32: %u Calculated CRC32: %u\n", name, Status, CRC32, CalcCRC32));
+      continue;
+    }
+
+    VarName = AllocatePool (VarList->NameSize);
+    if (VarName == NULL) {
+      Status = EFI_OUT_OF_RESOURCES;
+      return Status;
+    }
+
+    // Copy non-16 bit aligned name to VarName, not using StrCpyS functions as they assert on non-16 bit alignment
+    CopyMem(VarName, name, VarList->NameSize);
+
+    // write variable directly to var storage
+    Status = gRT->SetVariable(
+                      VarName,
+                      Guid,
+                      Attributes,
+                      VarList->DataSize,
+                      Data
+    );
+
+    FreePool(VarName);
+  }
+
+  return Status;
+}
+
+/**
+  It is not supported to get the full Runtime Setting list. Instead,
+  each individual Runtime Settings Provider can fetch the setting.
+
+  @param This       Setting Provider
+  @param ValueSize  IN=Size of location to store value
+                    OUT=Size of value stored
+  @param Value      Output parameter for the setting value.
+                    The type and size is based on the provider type
+                    and must be allocated by the caller.
+  @retval EFI_UNSUPPORTED   This is not supported for Runtime Settings
+**/
+EFI_STATUS
+EFIAPI
+RuntimeDataGet (
+  IN CONST DFCI_SETTING_PROVIDER  *This,
+  IN OUT   UINTN                  *ValueSize,
+  OUT      UINT8                  *Value
+  )
+{
+  return EFI_UNSUPPORTED;
+}
+
+/**
+  Default values of Runtime Settings are not supported.
+
+  @param This           Setting Provider
+  @param ValueSize      IN=Size of location to store value
+                        OUT=Size of value stored
+  @param DefaultValue   Output parameter for the settings default value.

```

Comment:
What is this?

Reply:
Copy/paste from the Conf side, where it also has the wrong variable name :)

Reply:
Fixed in all locations.